tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassloaderTestRunner.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * @since 4.0
+ */
+public class ArtifactClassloaderTestRunner extends Suite
+{
+    private static final List<Runner> NO_RUNNERS = emptyList();
+    private final ArrayList<Runner> runners = new ArrayList<>();
+
+    /**
+     * Creates a Runner to run {@code klass}
+     *
+     * @param klass
+     * @throws InitializationError if the test class is malformed.
+     */
+    public ArtifactClassloaderTestRunner(final Class<?> klass) throws Throwable
+    {
+        super(createTestClassUsingClassLoader(klass), NO_RUNNERS);
+        Class<? extends Runner> runnerClass = BlockJUnit4ClassRunner.class;
+        RunnerDelegateTo runnerDelegateTo = klass.getAnnotation(RunnerDelegateTo.class);
+        if (runnerDelegateTo != null)
+        {
+            runnerClass = runnerDelegateTo.value();
+        }
+        runners.add(runnerClass.cast(runnerClass.getConstructor(Class.class).newInstance(getTestClass().getJavaClass())));
+    }
+
+    @Override
+    protected List<Runner> getChildren() {
+        return runners;
+    }
+
+    @Override
+    public Description getDescription() {
+        // Just to avoid having duplicated name in IDEA/Eclipse JUnit panel view due to this is a Suite, we always have only one delegate/child runner.
+        return runners.get(0).getDescription();
+    }
+
+
+    protected void runChild(final Runner runner, final RunNotifier notifier) {
+        withContextClassLoader(getTestClass().getJavaClass().getClassLoader(), () -> runner.run(notifier));
+    }
+
+    /**
+     * @param klass
+     * @return the {@link ClassLoader} that would be used to run the test. This way the test will be isolated and it will behave
+     * similar as an application running in a Mule standalone container.
+     * @throws Throwable
+     */
+    private static Class<?> createTestClassUsingClassLoader(Class<?> klass) throws Throwable
+    {
+        // Initializes utility classes
+        ClassPathURLsProvider classPathURLsProvider = new DefaultClassPathURLsProvider();
+        MavenDependenciesResolver mavenDependenciesResolver = new DependencyGraphMavenDependenciesResolver();
+        MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping = new MuleMavenMultiModuleArtifactMapping();
+        ClassLoaderRunnerFactory classLoaderRunnerFactory = new MuleClassLoaderRunnerFactory();
+        ClassPathClassifier classPathClassifier = new MuleClassPathClassifier();
+
+        // Does the classification and creation of the isolated ClassLoader
+        ArtifactUrlClassification artifactUrlClassification = classPathClassifier.classify(new DefaultClassPathClassifierContext(klass, classPathURLsProvider.getURLs(),
+                                                                                           mavenDependenciesResolver.buildDependencies(), mavenMultiModuleArtifactMapping));
+        ClassLoaderTestRunner classLoaderTestRunner = classLoaderRunnerFactory.createClassLoader(klass, artifactUrlClassification);
```

Comment:
Not for this PR, but create a issue in order to improve the runer to avoid the need of having to create a new classLoader on every test. All the tests in a given module will have the same classLoader, so it seems non performant to create a new one every time.